bolt/lib/Target/X86/X86MCPlusBuilder.cpp

```diff
@@ -1932,6 +1932,19 @@ class X86MCPlusBuilder : public MCPlusBuilder {
     //    = R_X86_64_PC32(Ln) + En - JT
     //    = R_X86_64_PC32(Ln + offsetof(En))
     //
+    auto isRIPRel = [&](X86MemOperand &MO) {
+      // NB: DispExpr should be set
+      return MO.DispExpr != nullptr &&
```

Comment:
this is adding this extra check, `MO.DispExpr != nullptr` -- why?

Reply:
No, it's actually just moving forward because it's the one that's important :)
And judging by the fact that you noticed it, the change achieved its goal.